### DISCUSSION TITLE
Normal Hex Color Formats

### DISCUSSION THREAD

**Discussion by NyhmsQuest** [Zrips/CMI#7887] [Enhancement]
Would it be possible to add #123123 and or &#123123 as working hex formats? Literally every other plugin does this and it would just be nice for consistency for it to be same everywhere. It's also a lot easier and quicker to write out than using { }

**Comment by 456dev** [comment]
Those other formats don't allow for gradients or named text colours, so it's unlikely to change
`{#red>}hello world!{#atoll<}`

**Reply by NyhmsQuest** [reply]
I’m not saying to remove the current format, just to allow more formats like the examples I wrote.

**Reply by 456dev** [reply]
How would https://example.com/path#123455-ananchor parse with #[hexcode]
It could be an acceptable trade off. Maybe it could be a config option?

And &#[hexcode] would probably break current legacy parsing, as that would only expect 1 extra character in front of it. Although if it's parsed separately, maybe a find replace to the actual format?

The only one I can think of would probably be minimessage, `<#123456>` l, but that would still need a rework in how parsing is done, and is also tag based

**Reply by NyhmsQuest** [reply]
I don't know the technical reason as to why not have it, all I know is that every other plugin uses either #123123 and or &#123123 and using the same formats for all plugins would make life easier for everyone. And I really don't see how it would be bad to add more options for a user of a plugin. More options is always best.

**Comment by Zrips** [comment]
Will add somewhat support with next CMILib update, tho keep in mind that due to its simplicity it will only support hex color codes as #6f6 or #6f6f6f, but will not support gradients or named colors which will still require that more complex format to be used. Disabled by default, so you will need to enable it in CMILib config file. Can guarantee that it will work perfectly in all situations as its more prone to errors (false/positive detections) in this simpler format. If you encounter any issues with it, let me know in CMI issue section.

**Reply by NyhmsQuest** [reply]
Perfect! Thank you so much. I never used named colors or really the built in gradient system you made. Typically I would use https://rgb.birdflop.com/ to create Gradients etc so this would be a very welcome change.

**Reply by NyhmsQuest** [reply]
Would it be possible to also add &#6f6f6f or is that not as easy as just adding #6f6f6f ?

**Reply by Zrips** [reply]
Gradients are as simple as {#084CFB>}birdflop.com{#ADF3FD<} which would produce your desired results. 
Adding &# is simple as its even better format than that clean one, as this one automatically rules out quite few false/positives. Will add that one too as I'm on it already.
